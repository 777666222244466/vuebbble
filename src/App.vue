<template>
  <div id="app">
    <vuebbble token="ac522ca5908d196031bba38306459d5cb341e7511c60ddd8dd668d2d4c1748f5" user="robbinfellow"></vuebbble>
  </div>
</template>

<script>
import Vuebbble from './components/Vuebbble.vue'

export default {
  name: 'app',
  components: {
    Vuebbble
  }
}
</script>
