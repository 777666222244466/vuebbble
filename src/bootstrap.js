window._ = require('lodash');
window.Vue = require('vue');
/**
 * Uncomment below when compiling to production
 */
// Vue.config.devtools = false
// Vue.config.debug = false
// Vue.config.silent = true

window.axios = require('axios');