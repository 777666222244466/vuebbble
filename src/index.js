import Vuebbble from './components/Vuebbble.vue'

const plugin = {
  install: Vue => {
    Vue.component('vuebbble', Vuebbble)
  }
}

// Automatic installation if Vue has been added to the global scope.
if (typeof window !== 'undefined' && window.Vue) {
    window.Vue.use(plugin)
}

export default plugin
export { Vuebbble }
