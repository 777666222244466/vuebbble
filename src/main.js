require('./bootstrap');
  
Vue.component('vuebbble', require('./components/Vuebbble.vue'));