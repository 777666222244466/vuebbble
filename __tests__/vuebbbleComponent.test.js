import { Vuebbble } from '../src';
import Vue from 'vue/dist/vue.js';

describe('Vuebbble', () => {
    Vue.component('vuebbble', Vuebbble);

    beforeEach(() => {
        document.body.innerHTML = `
            <div id="app">
                <vuebbble token="8828b898f4e2cec33f65f5342ba736b5e35e70f03681aa7435ad26c3b1830847" user="robbinfellow"></vuebbble>
            </div>
        `;
    });

    it('can mount', async () => {
        await createVm();

        expect(document.body.innerHTML).toMatchSnapshot();
    });
});

async function createVm() {
    const vm = new Vue({
        el: '#app',
    });

    await Vue.nextTick(() => {});

    return { app: vm, component: vm.$children[0] };
}
