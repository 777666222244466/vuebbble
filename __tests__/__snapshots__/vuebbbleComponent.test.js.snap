// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Vuebbble can mount 1`] = `
"
            <div id=\\"app\\"><div class=\\"vuebbble\\"> <!----></div></div>
        "
`;
